### PR TITLE
Added `dvclive.set_step`

### PR DESCRIPTION
**Pull request by daavoo**
`dvclive` P.R: https://github.com/iterative/dvclive/pull/157

### CODE REVIEW

**Comment by dberenbaum** [comment]
Should this be marked as draft until the PR is merged?

**Comment by daavoo** [comment]
> Should this be marked as draft until the PR is merged?

I though that the label `⏳ awaiting-dvc-merge` was used for the same purpose

**Comment by dberenbaum** [comment]
Sorry about that! I didn't notice the label.

**Comment by jorgeorpinel** [comment]
> Let's hold off on review until we decide on the behavior of the core PR.

@dberenbaum so https://github.com/iterative/dvclive/pull/157 is not approved?

**Comment by dberenbaum** [comment]
Sorry, I approved with a comment, but the reply from @daavoo opened up a new set of questions on the behavior, so I'd say it's not approved until we decide on the expected behavior from those comments.

File: content/docs/dvclive/api-reference/get_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/get_step.md:28]: This is a bit cryptic/ mathematical. How about something like

> ... increases the `step` count by 1 (one). You can use `dvclive.set_step()` to increment the step count by any other number.

Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/get_step.md:28]: `dvclive.set_step()` needs an auto-link entry BTW 🙂

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/set_step.md:32]:
```suggestion
Given the [Usage](#usage) code snippet above, the
```

**Review by jorgeorpinel — COMMENTED** [review]
OK well I already reviewed anyway 😬 but feel free to request another review. Thanks

File: content/docs/dvclive/api-reference/get_step.md
Comment (daavoo) [reply content/docs/dvclive/api-reference/get_step.md:28]: > dvclive.set_step() needs an auto-link entry BTW 🙂

It should be working now (it works for me in the deployment)

**Comment by jorgeorpinel** [comment]
Ping @daavoo 

Also, there's a bad link, something strange with the auto-linking it seems: https://github.com/iterative/dvc.org/pull/2823/checks?check_run_id=3607835623

File: content/docs/dvclive/api-reference/set_step.md
Comment (dberenbaum) [inline content/docs/dvclive/api-reference/set_step.md:18]: I thought it's not necessary to have both `set_step` and `next_step`?

Comment (daavoo) [reply content/docs/dvclive/api-reference/set_step.md:18]: Correct

File: content/docs/dvclive/api-reference/set_step.md
Comment (daavoo) [reply content/docs/dvclive/api-reference/set_step.md]: Updated the message and code block as the suggested change is not actually true

**Comment by daavoo** [comment]
> Ping @daavoo
> 
> Also, there's a bad link, something strange with the auto-linking it seems: https://github.com/iterative/dvc.org/pull/2823/checks?check_run_id=3607835623

I have no idea why that check is failing. I tested de auto-linking locally and the line where the `link-check-deploy.yml` fails is not matched by the auto linker 🤷

**Comment by shcheklein** [comment]
@rogermparent @julieg18  - also here - do we know why does link checker fail?

File: content/docs/dvclive/api-reference/set_step.md
Comment (shcheklein) [inline content/docs/dvclive/api-reference/set_step.md:50]: `$ cat`

File: content/docs/dvclive/api-reference/set_step.md
Comment (shcheklein) [inline content/docs/dvclive/api-reference/set_step.md:29]: minor: should be `DVC integration`?

File: content/docs/dvclive/api-reference/set_step.md
Comment (shcheklein) [inline content/docs/dvclive/api-reference/set_step.md:32]: features -> effects? outcomes? side effects?

File: content/docs/dvclive/api-reference/next_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/next_step.md:44]: "Outcomes" sounds more high level 🙂 
```suggestion
have additional effects.
```

File: content/docs/dvclive/api-reference/get_step.md
Comment (dberenbaum) [inline content/docs/dvclive/api-reference/get_step.md:24]:
```suggestion
Each call to `dvclive.next_step()` increases the `step` count by 1 (one).
```

File: content/docs/dvclive/api-reference/next_step.md
Comment (dberenbaum) [inline content/docs/dvclive/api-reference/next_step.md:4]: What is being configured about next_step behavior in init?

File: content/docs/dvclive/api-reference/next_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/next_step.md:5]: This info. should be down in the description, I think.

Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/next_step.md:5]: In fact it is (not linked directly but still should be OK) so we can remove this here, IMO.
```suggestion
```

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/set_step.md:24]:
```suggestion
DVCLive uses the `step` value to track the progress of each metric logged with
`dvclive.log()`. You can use `dvclive.set_step()` to set `step` to any value.
```

File: content/docs/dvclive/api-reference/next_step.md
Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/next_step.md:4]: I think this is explained in the description @dberenbaum but not sure the info is complete (I'd remove this sentence from the intro paragraph).

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/set_step.md:3]: > updates `step` with the given value

Does it increase it by the given value? Sets it to the given value? Or defines the "step value" for which `step` will be increases with `next_step()`? Let's be more specific here to avoid speculation. Thanks

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/set_step.md:23]: > set `step` to any value.

Can there be invalid values? Like going backwards, repeating numbers, negatives, floats, etc.

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/set_step.md:26]: > in between `dvclive.set_step()` calls

Could you `set_step` and then `next_step`?

```suggestion
Each metric logged in between `dvclive.set_step()` (or `dvclive.next_step()`)
calls will be associated to the provided `step` value.
```

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/set_step.md:31]:
```suggestion
additional effects.
```

**Review by jorgeorpinel — APPROVED** [review]
A few minor clarifications still needed but otherwise LGTM. Thanks @daavoo !

File: content/docs/dvclive/api-reference/set_step.md
Comment (rogermparent) [inline content/docs/dvclive/api-reference/set_step.md:34]: This one actually looks broken, I believe it should be `/doc/dvclive/dvclive-with-dvc#html-report`

File: content/docs/dvclive/api-reference/set_step.md
Comment (rogermparent) [inline content/docs/dvclive/api-reference/set_step.md:39]: Similarly, this should be `/doc/dvclive/dvclive-with-dvc#checkpoints`, since the `dvclive` docs have no `user-guide` section.

File: content/docs/dvclive/api-reference/set_step.md
Comment (rogermparent) [inline content/docs/dvclive/api-reference/set_step.md:15]: The link checker is seeing this as a link for some reason, that's on us and can be ignored.

**Review by rogermparent — COMMENTED** [review]
One is a link check error on the regex side, but the other two links are actually broken.

File: content/docs/dvclive/api-reference/set_step.md
Comment (daavoo) [reply content/docs/dvclive/api-reference/set_step.md:3]: This:
> Sets it to the given value directly

Updated to be more specific

File: content/docs/dvclive/api-reference/set_step.md
Comment (daavoo) [reply content/docs/dvclive/api-reference/set_step.md:23]: We don't currently have any restrictions implemented so technically it can be any of those.

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/set_step.md]: Thanks!

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/set_step.md:23]: Not floats though, right? `step: int`

File: content/docs/dvclive/api-reference/set_step.md
Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/set_step.md:15]: Haha

```js
[0, 10, 20]:
    dvclive.set_step(step)
```
looks like an md reference link 🤦 

Ideally the link checker should ignore everything inside md code blocks, I think.

Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/set_step.md:15]: p.s. a quick fix would be to start that regex with `^\[`.

**Pull request merged**
